dask/highlevelgraph.py

```diff
@@ -780,6 +780,30 @@ def cull(self, keys: set) -> "HighLevelGraph":
 
         return HighLevelGraph(ret_layers, ret_dependencies, ret_key_deps)
 
+    def cull_layers(self, layers: Iterable[str]) -> "HighLevelGraph":
+        """Return a new HighLevelGraph with only the given layers and their
+        dependencies. Internally, layers are not modified.
+
+        This is a variant of :meth:`HighLevelGraph.cull` which is much faster and does
+        not risk creating a collision between two layers with the same name and
+        different content when two culled graphs are merged later on.
+
+        Returns
+        -------
+        hlg: HighLevelGraph
+            Culled high level graph
+        """
+        to_visit = set(layers)
+        ret_layers = {}
+        ret_dependencies = {}
+        while to_visit:
+            k = to_visit.pop()
+            ret_layers[k] = self.layers[k]
+            ret_dependencies[k] = self.dependencies[k]
+            to_visit |= ret_dependencies[k] - ret_dependencies.keys()
+
+        return HighLevelGraph(ret_layers, ret_dependencies)
+
```

Comment:
See https://github.com/pydata/xarray/pull/4965 for real-life usage of this new method

Reply:
cc @madsbk  for visibility about this new `cull_layers` method